### ISSUE TITLE
Usage of `isSubsequenceOf` causing false positives in the NFData check

### ISSUE THREAD

**Issue by 0xd34df00d** [CRogers/should-not-typecheck#18]
Hi! Great library, but I've come across a tiny issue.

Basically the check for whether `NFData` is implemented fires on the following deferred type error message, seemingly due to `isSubsequenceOf` in the corresponding check in the `shouldNotTypecheck` function:
```
/home/d34df00d/Programming/can-i-haz/test/TypecheckFailures.hs:11:10: error:
           • Couldn't match type ‘'Data.Path.NotFound’
                            with ‘'Data.Path.Found path0’
               arising from a use of ‘Control.Monad.Reader.Has.$dmextract’
           • In the expression:
               Control.Monad.Reader.Has.$dmextract @(BarEnv) @(FooEnv)
             In an equation for ‘extract’:
                 extract = Control.Monad.Reader.Has.$dmextract @(BarEnv) @(FooEnv)
             In the instance declaration for ‘Has BarEnv FooEnv’
       (deferred type error)
```

Looks like it would be gone if `isInfixOf` is used instead. If you're happy with this change, then I'm happy to make a PR!

**Comment by 0xd34df00d** [comment]
Alright, just noticed `HEAD` already uses `isInfixOf`. I think this one may then be closed (perhaps asking for a new version release :)).
